src/client-python/reportclient/internal/bz_connection.py

```diff
@@ -352,10 +354,22 @@ def bug_search(self, query: Dict) -> List:
         self.logger.debug('-- %s', inspect.getframeinfo(inspect.currentframe()).function)
         params = self.params.copy()
         params.update(query)
-        response = requests.get(os.path.join(self.url, 'rest.cgi/bug'),
-                                headers=self.headers,
-                                params=params,
-                                verify=self.verify)
+
+        # Configure requests to retry with delays (See: rhbz#2208742)
+        session = requests.Session()
+        retry = Retry(connect=5, backoff_factor=0.5)
+        adapter = HTTPAdapter(max_retries=retry)
+        session.mount('http://', adapter)
+        session.mount('https://', adapter)
+        try:
+            response = session.get(os.path.join(self.url, 'rest.cgi/bug'),
```

Comment:
This should be using `urllib.parse.urljoin()` rather than `os.path.join()`.

Reply:
Fixed ;)